backend/internal/controller/adapter/dispatcher_test.go

```diff
@@ -0,0 +1,141 @@
+package adapter
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetMessageCategory(t *testing.T) {
+
+	type args struct {
+		name             string
+		message          dispatcherMsg
+		expectedCategory MsgCategory
+	}
+	tests := []args{
+		{
+			name: "initial message, should return MsgInitial",
+			message: dispatcherMsg{
+				MsgType: "INITIAL",
+				Goal:    0,
+				X:       0,
+				Y:       0,
+			},
+			expectedCategory: MsgInitial},
+		{
+			name: "goal message, should return MsgGoal",
+			message: dispatcherMsg{
+				MsgType: "",
+				Goal:    1,
+				X:       0,
+				Y:       0,
+			},
+			expectedCategory: MsgGoal},
+		{
+			name: "position message, should return MsgNone",
+			message: dispatcherMsg{
+				MsgType: "",
+				Goal:    0,
+				X:       4,
+				Y:       2.22,
+			},
+			expectedCategory: MsgNone},
+		{
+			name: "unexpected message, should return MsgNone",
+			message: dispatcherMsg{
+				MsgType: "SOMETHING",
+				Goal:    0,
+				X:       15,
+				Y:       0},
+			expectedCategory: MsgNone},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			category := tt.message.getMessageCategory()
+			assert.Equal(t, category, tt.expectedCategory)
+		})
+	}
+}
+
+func TestUnpack(t *testing.T) {
+	type args struct {
+		name           string
+		msgIn          dispatcherMsg
```

Comment:
A small hint, if you have decided not to use abbreviated words, e.g. `message` rather than `msg`, it is a good idea to stick to one idea in the whole file.